volume/drivers/nfs/nfs.go

```diff
@@ -319,14 +416,18 @@ func (d *driver) Delete(volumeID string) error {
 		return err
 	}
 
-	// Delete the simulated block volume
-	os.Remove(v.DevicePath)
+	if v.GetState() == api.VolumeState_VOLUME_STATE_ATTACHED {
+		return fmt.Errorf("Volume is still attached and cannot be deleted")
+	}
 
 	nfsVolPath, err := d.getNFSVolumePath(v)
 	if err != nil {
 		return err
 	}
 
+	// Delete the simulated block volume
+	os.Remove(nfsVolPath + nfsBlockFile)
```

Comment:
Maybe check for errors on all os.Remove*? For pesky issues with permissions or the file system in general